Gems/ScriptCanvas/Code/Editor/Components/EditorGraph.cpp

```diff
@@ -1141,7 +1141,7 @@ namespace ScriptCanvasEditor
             {
                 if (slot->IsVariableReference())
                 {
-                    return true;
+                    return !slot->IsUserAdded();
```

Comment:
Should a new named function be introduced? we do this testing more than once, perhaps "IsFunctionVariableReference" or something along those lines?

```bool IsFunctionVariableReference() const { return slot->IsVariableReference() && slot->IsUserAdded(); }```

Reply:
There's a wider, architectural change that needs to happen. Probably won't add a new function in here, as usually the test is is-reference? and then is-user-added? and different actions occur on the second if-branch.